services/galley/src/Galley/API/Create.hs

```diff
@@ -254,31 +261,58 @@ checkCreateConvPermissions lusr newConv Nothing allUsers = do
+      -- regular conversations, therefore we check for 'AddRemoveConvMember' only if
+      -- there are going to be more than two users in the conversation.
+      -- FUTUREWORK: We keep this permission around because not doing so will break backwards
+      -- compatibility in the sense that the team role 'partners' would be able to create group
+      -- conversations (which they should not be able to).
+      -- Not sure at the moment how to best solve this but it is unlikely
+      -- we can ever get rid of the team permission model anyway - the only thing I can
+      -- think of is that 'partners' can create convs but not be admins...
+      -- this only applies to proteus conversations, because in MLS we have proper 1:1 conversations,
+      -- so we don't allow an external partner to create an MLS group conversation at all
+      when (length allUsers > 1 || newConv.newConvProtocol == BaseProtocolMLSTag) $ do
+        void $ permissionCheck AddRemoveConvMember zusrMembership
+
   convLocalMemberships <- mapM (E.getTeamMember convTeam) (ulLocals allUsers)
   ensureAccessRole (accessRoles newConv) (zip (ulLocals allUsers) convLocalMemberships)
-  -- In teams we don't have 1:1 conversations, only regular conversations. We want
-  -- users without the 'AddRemoveConvMember' permission to still be able to create
-  -- regular conversations, therefore we check for 'AddRemoveConvMember' only if
-  -- there are going to be more than two users in the conversation.
-  -- FUTUREWORK: We keep this permission around because not doing so will break backwards
-  -- compatibility in the sense that the team role 'partners' would be able to create group
-  -- conversations (which they should not be able to).
-  -- Not sure at the moment how to best solve this but it is unlikely
-  -- we can ever get rid of the team permission model anyway - the only thing I can
-  -- think of is that 'partners' can create convs but not be admins...
-  -- this only applies to proteus conversations, because in MLS we have proper 1:1 conversations,
-  -- so we don't allow an external partner to create an MLS group conversation at all
-  when (length allUsers > 1 || newConv.newConvProtocol == BaseProtocolMLSTag) $ do
-    void $ permissionCheck AddRemoveConvMember zusrMembership
-
   -- Team members are always considered to be connected, so we only check
   -- 'ensureConnected' for non-team-members.
   ensureConnectedToLocals (tUnqualified lusr) (notTeamMember (ulLocals allUsers) (catMaybes convLocalMemberships))
   ensureConnectedToRemotes lusr (ulRemotes allUsers)
 
+ensureCreateChannelPermissions ::
+  forall r.
+  ( Member (ErrorS OperationDenied) r,
+    Member (Input Opts) r,
+    Member TeamFeatureStore r,
+    Member (ErrorS NotATeamMember) r
+  ) =>
+  NewConv ->
+  Maybe TeamMember ->
+  Sem r ()
+ensureCreateChannelPermissions newConv (Just tm) = do
+  case newConv.newConvGroupConvType of
```

Comment:
Case on newConvGroupConvType twice here. Can this function be refactored to remove the need to case again?